### ISSUE TITLE
DTLS maximum record size

### ISSUE THREAD

**Issue by martinthomson** [gloinul/draft-westerlund-tsvwg-dtls-over-sctp-bis#132]
https://gloinul.github.io/draft-westerlund-tsvwg-dtls-over-sctp-bis/draft-ietf-tsvwg-dtls-over-sctp-bis.html#section-3.5-1 says:

> [...] DTLS can send maximum sized DTLS Records.

This is almost certainly bad advice.  Some DTLS implementations can't handle arbitrary-sized records, so you might find that things fail if you handshake with 2<sup>14</sup> bytes in a record.  For performance reasons, you probably want smaller records in many cases.  Aligning record sizes to the MTU can make sense as well.  

See also my recommendations about RFC 8449 in #130, which addresses a resource protection issue.  In addition to that, you might recommend keeping record sizes small until you know what the size limit is.

**Comment by gloinul** [comment]
Yes, we will have to consider RFC 8449. I can understand how DTLS record size are not expected for really large sizes but, I would have thought this would have been a lesser issue due to the TLS origin where larger records should be common. 

However, using IP MTU sized DTLS record just increases the overhead. SCTP provides the re-assembly layer and also the exact data portion per DATA chunk is not necessary static. So that aspect appear to have its downsides too.

**Comment by gloinul** [comment]
So @teiclap and I did discuss this. I will assume that you mean that Martin meant the first above that implementations don't support larger than a particular size that is less than 2^14. 

Based on how DTLS/SCTP is intended to be used and for interoperability I think the reasonable way forward here is to default to require the DTLS stack do support max sized (2^14) bytes DTLS records. Implementation that want to negotiate a lower maximum DTLS records SHALL use RFC 8449 to accomplish that. Thus, if one want to do any small records for the reasons discussed in the text that was brought up in issue #130 it will also require RFC 8449. 

Considering that RFC 6083 has required full size DTLS record support also, and that RFC 8449 is additional implementation requirement we propose to not require RFC 8449 support, and if an endpoint is not capable of supporting 2^14 DTLS records and the peer does not support RFC 8449 one will be at an impasse and will have to terminate the SCTP association due to non-interoperability. 

We also note that to avoid massive implementation complexities the implementations MUST NOT change the maximum DTLS Record size negotiated using the RFC 8449 TLS extension. This as DTLS/SCTP needs to be aware of this limitation to control its data fragmentation process. And forcing a change of the data fragmentation mid message that such a rekeying could result in would result in some complexities.

**Comment by martinthomson** [comment]
This seems reasonable.  I would expect some level of recommendation about 8449 support so that you don't get the interop failure.

**Comment by gloinul** [comment]
I think SHOULD/RECOMMENDED support of RFC8449 for general deployments of DTLS/SCTP and some usage of this could simply mandate RFC8449 support. But will think more about this as we draft spec text.

**Comment by teiclap** [comment]
Related to DTLS record size, the section 3.5 describes only the PMTU, I think that the current text is proper.
The description of the interaction between DTLS/SCTP and 8449 has been rewritten in my PR for issue #130
